udp_writer_guide.md

```diff
@@ -0,0 +1,63 @@
+## Guide for writing openEO User Defined Processes
+
+User defined processes (UDPs) are one out of two standardized options that APEx offers to publish algorithms as a service.
+This guide gives some concrete steps and guidelines to ensure that your UDP works well for your users. These guidelines are
+written with APEx in mind, but can also serve as a general guide for openEO UDPs. Where needed, recommendations & choices 
+are made to increase uniformity across the APEx algorithm catalog.
+
+
+### Organizing your code
+
+A UDP is simply a parametrized version of an openEO workflow, and as such we recommend that you use the same code to develop
+and test your algorithm, as you use to generate the UDP json. This ensures that your UDP is functionally equivalent to your code.
+Your code remains your own, and you only need to export the JSON UDP definition to share it with APEx.
+
+It is however advisable for your UDP to link back to the git repository if relevant, making your open source code more 
+discoverable. 
+
+### How many UDPs should I write?
+
+Deciding on the granularity of your UDP is an important aspect of making your algorithm usable. There is no need to try
+and fit all possible use cases into a single UDP. Instead, consider to define pieces of functionality that can work with
+a limited set of parameters, and write a single UDP for each piece.
+
+
+### Parameter conventions
+
+#### Spatial filtering
+
+Any algorithm requires spatial filtering, so we can make the life of our users easier if we all use the same naming.
+
+In other openEO processes, the spatial filtering is called `spatial_extent`, and we recommend to stick to that name as 
+long as you don't have more specific names.
+
+If you pass on the `spatial_extent' parameter to all `load_collection` processes in your UDP, then it is also allowed to
+perform filtering using vector data. This conveniently allows for advanced spatial filtering use cases!
+
+#### Temporal filtering
+
+For UDP's that support arbitrary temporal ranges, the `temporal_extent` parameter from `load_collection` is the recommended option.
+
+Many cases also require a time range with a fixed length. In such a case, you can allow to specify only the start or end
+date, and use the `date_shift` process to construct the second date in the temporal interval, ensuring a fixed length. This
```

Comment:
this might be already a bit more complex for entry users?

Reply:
(the use of date_shift)

Reply:
The target audience here is people trying to produce a high quality UDP within the context of an ESA project.
So we should not necessarily limit it to 'easy' things.
We can however add some examples as soon as I have any. That would probably go in a separate PR.

Reply:
then I would at least add a link here to the actual data_shift process